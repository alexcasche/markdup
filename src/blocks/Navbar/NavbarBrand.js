import { RouterLink } from 'elements'
import { shade, tint } from 'polished'

const NavbarBrand = RouterLink.extend`
  color: ${props => props.theme.colors.grey};
  font-family: ${props => props.theme.text.fontLogo};
  font-size: 20px;
  letter-spacing: 3px;
  text-transform: uppercase;
  text-decoration: none;
  transition: color 0.15s ease-in-out;
  span {
    color: ${props => shade(0.8, props.theme.colors.grey)};
    transition: inherit;
  }
  &:hover {
    color: ${props => tint(0.25, props.theme.colors.grey)};
    span {
      color: ${props => tint(0.65, props.theme.colors.grey)};
    }
  }
`

export default NavbarBrand
