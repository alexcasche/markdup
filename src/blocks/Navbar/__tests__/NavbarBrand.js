import { BrowserRouter } from 'react-router-dom'
import NavbarBrand from '../NavbarBrand'

describe('NavbarBrand Element', () => {
  it('renders correctly', () => {
    const tree = render(
      <BrowserRouter>
        <NavbarBrand to="/">
          <span>markd</span>up
        </NavbarBrand>
      </BrowserRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
