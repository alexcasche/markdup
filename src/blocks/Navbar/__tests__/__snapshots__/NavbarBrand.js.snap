// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarBrand Element renders correctly 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  color: #ABADB3;
  font-family: "Raleway",sans-serif;
  font-size: 20px;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c0:hover {
  cursor: pointer;
}

.c0 span {
  color: #888a8f;
  -webkit-transition: inherit;
  transition: inherit;
}

.c0:hover {
  color: #eaeaec;
}

.c0:hover span {
  color: #c8c9cd;
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
>
  <span>
    markd
  </span>
  up
</a>
`;
